src/prefect/client/schemas/actions.py

```diff
@@ -538,6 +545,31 @@ def convert_value(value: Any) -> Any:
             )
         return values
 
+    @field_serializer("parameters", when_used="json")
+    def serialize_parameters(self, value: dict[str, Any]) -> dict[str, Any]:
+        """Serialize datetime types as ISO strings instead of timestamps.
+
+        PrefectBaseModel has ser_json_timedelta='float' to serialize timedeltas as floats,
+        but this also causes datetime/date/time to serialize as timestamps. This serializer
+        overrides that behavior for datetime types while preserving float serialization for
+        timedeltas.
+        """
+
+        def convert_temporal(v: Any) -> Any:
+            if isinstance(v, datetime.datetime):
+                return v.isoformat()
+            elif isinstance(v, datetime.date):
+                return v.isoformat()
+            elif isinstance(v, datetime.time):
+                return v.isoformat()
+            elif isinstance(v, dict):
+                return {k: convert_temporal(val) for k, val in v.items()}
+            elif isinstance(v, list):
+                return [convert_temporal(item) for item in v]
+            return v
```

Comment:
You can collapse some of these checks.
```suggestion
            if isinstance(v, (datetime.datetime, datetime.date, datetime.time)):
                return v.isoformat()
            elif isinstance(v, dict):
                return {k: convert_temporal(val) for k, val in v.items()}
            elif isinstance(v, list):
                return [convert_temporal(item) for item in v]
            return v
```